Test: Run phan with github workflows

kesselb opened the pull request:
Signed-off-by: Daniel Kesselberg <mail@danielkesselberg.de>

MorrisJobke commented:
Nice ... was about to resurrect the one in drone: https://github.com/nextcloud/server/commit/a987f809e647122c246bd602132c9fe8f8dd1ec6

but this then can go.

kesselb commented:
I changed the level back to critical. Still some warnings to fix. Also SqlInjectionCheckerPlugin is commented. Not sure how to update it.

kesselb committed:
Help phan to understand code.

Signed-off-by: Daniel Kesselberg <mail@danielkesselberg.de>

MorrisJobke commented on apps/admin_audit/lib/AppInfo/Application.php:
```diff
@@ -143,8 +143,8 @@ private function groupHooks(ILogger $logger,
 
-		/** @var IGroupManager|Manager $groupManager */
 		$groupManager = $serverContainer->getGroupManager();
+		assert($groupManager instanceof GroupManager);
```
Nice one 👍

MorrisJobke commented on apps/provisioning_api/lib/AppInfo/Application.php:
```diff
@@ -58,8 +60,16 @@ public function __construct(array $urlParams = []) {
+
+			if ($user instanceof IUser) {
+				$groupManager = $server->getGroupManager();
+				assert($groupManager instanceof GroupManager);
```
I guess this is not needed, because the return type of the method is already specific enough, isn't it?

MorrisJobke replied:
if not then I would enhance the return code of the `IServerContainer` instead of adding it here.

kesselb replied:
I'm not to happy with `assert` and `@phan-var`. It's more a hack to suppress the warning. The issue itself (IGroupManager is injected but methods not part of the interface are requested) is not fixed.

kesselb replied:
> I guess this is not needed, because the return type of the method is already specific enough, isn't it?

No :( IGroupManager does not expose getSubAdmin and getGroupManager should return a IGroupManager instance. It works because getGroupManager always returns \OC\Group\Manager.

kesselb committed:
Annotate how query works

https://github.com/phan/phan/wiki/Generic-Types#indicating-a-method-creates-an-instance-of-the-passed-in-class-name
Signed-off-by: Daniel Kesselberg <mail@danielkesselberg.de>

kesselb committed:
Help phan to understand code.

Signed-off-by: Daniel Kesselberg <mail@danielkesselberg.de>

MorrisJobke commented:
I guess the fixes from here are fine, but the Phan part can go - see #22883

MorrisJobke commented:
> I guess the fixes from here are fine, but the Phan part can go - see #22883

Let me rebase this.

MorrisJobke commented:
Let's extract them step by step in separate PRs.

MorrisJobke commented:
This went all into #22888 #28890 and #28891. Can be closed.
